addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1732,6 +1733,36 @@ QUnit.test('first unseen message should be directly preceded by the new message
     );
 });
 
+QUnit.test('focus should be on textarea on manual send [REQUIRE FOCUS]', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.channel'].records = [{
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        name: "General",
+        uuid: 'channel20uuid',
+    }];
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+    document.querySelector('.o_ComposerTextInput_textarea').focus();
+    await afterNextRender(() => document.execCommand('insertText', false, "Dummy Message"));
+    await afterNextRender(() => {
+        document.querySelector('.o_Composer_buttonSend').click();
+    });
+
+    assert.containsN(document.body, '.o_Message', 1, "should display 1 message");
+    assert.ok(thread.composer.hasFocus, 'focus should be on textarea on manual send');
```

Comment:
This is a component test, should assert on DOM state  (composer hasClass `o-focused`) instead of model state